container/apptabs.go

```diff
@@ -18,670 +23,396 @@ import (
-func (c *AppTabs) ExtendBaseWidget(wid fyne.Widget) {
-	c.BaseWidget.ExtendBaseWidget(wid)
+func (t *AppTabs) ExtendBaseWidget(wid fyne.Widget) {
+	t.BaseWidget.ExtendBaseWidget(wid)
+}
+
+// Hide hides the widget.
+//
+// Implements: fyne.Widget
+func (t *AppTabs) Hide() {
+	if t.popUpMenu != nil {
+		t.popUpMenu.Hide()
+		t.popUpMenu = nil
+	}
+	t.BaseWidget.Hide()
 }
 
 // MinSize returns the size that this widget should not shrink below
-func (c *AppTabs) MinSize() fyne.Size {
-	c.BaseWidget.ExtendBaseWidget(c)
-	return c.BaseWidget.MinSize()
+func (t *AppTabs) MinSize() fyne.Size {
+	t.BaseWidget.ExtendBaseWidget(t)
+	return t.BaseWidget.MinSize()
 }
 
-// Remove tab by value
-func (c *AppTabs) Remove(item *TabItem) {
-	for index, existingItem := range c.Items {
-		if existingItem == item {
-			c.RemoveIndex(index)
-			break
-		}
-	}
+// Remove tab by value.
+func (t *AppTabs) Remove(item *TabItem) {
+	removeItem(t, item)
+	t.Refresh()
 }
 
-// RemoveIndex removes tab by index
-func (c *AppTabs) RemoveIndex(index int) {
-	c.SetItems(append(c.Items[:index], c.Items[index+1:]...))
+// RemoveIndex removes tab by index.
+func (t *AppTabs) RemoveIndex(index int) {
+	removeIndex(t, index)
+	t.Refresh()
 }
 
-// SetItems sets the container’s items and refreshes.
-func (c *AppTabs) SetItems(items []*TabItem) {
-	c.Items = items
-	if l := len(c.Items); c.current >= l {
-		c.current = l - 1
-	}
-	c.Refresh()
+// Select sets the specified TabItem to be selected and its content visible.
+func (t *AppTabs) Select(item *TabItem) {
+	selectItem(t, item)
+	t.Refresh()
+}
+
+// SelectIndex sets the TabItem at the specific index to be selected and its content visible.
+func (t *AppTabs) SelectIndex(index int) {
+	selectIndex(t, index)
+	t.Refresh()
 }
 
 // SelectTab sets the specified TabItem to be selected and its content visible.
-func (c *AppTabs) SelectTab(item *TabItem) {
-	for i, child := range c.Items {
+//
+// Deprecated: Use `AppTabs.Select(*TabItem)` instead.
+func (t *AppTabs) SelectTab(item *TabItem) {
+	for i, child := range t.Items {
 		if child == item {
-			c.SelectTabIndex(i)
+			t.SelectTabIndex(i)
 			return
 		}
 	}
 }
 
 // SelectTabIndex sets the TabItem at the specific index to be selected and its content visible.
-func (c *AppTabs) SelectTabIndex(index int) {
-	if index < 0 || index >= len(c.Items) || c.current == index {
+//
+// Deprecated: Use `AppTabs.SelectIndex(int)` instead.
+func (t *AppTabs) SelectTabIndex(index int) {
+	if index < 0 || index >= len(t.Items) || t.current == index {
 		return
 	}
-	c.current = index
-	c.Refresh()
+	t.current = index
+	t.Refresh()
 
-	if c.OnChanged != nil {
-		c.OnChanged(c.Items[c.current])
+	if t.OnChanged != nil {
+		t.OnChanged(t.Items[t.current])
 	}
 }
 
-// SetTabLocation sets the location of the tab bar
-func (c *AppTabs) SetTabLocation(l TabLocation) {
-	c.tabLocation = l
-	c.Refresh()
+// Selection returns the currently selected TabItem.
+func (t *AppTabs) Selection() *TabItem {
```

Comment:
It should be `Selected` - we have not implemented (nor finalised) the multi-select enabling future API for selection

Reply:
That is not what #1395 says, and tabbers will never support multi-select

Reply:
`Selection` in the (not yet implemented) proposal is for multiselect - so see above for recommendation

Reply:
Done